src/main/java/org/newrelic/nrjmx/v2/JMXFetcher.java

```diff
@@ -134,6 +133,10 @@ public void disconnect(long timeoutMs) throws JMXError, JMXConnectionError {
      * @throws JMXConnectionError JMX connection related exception
      */
     public void disconnect() throws JMXConnectionError {
+        if (Thread.interrupted()) {
```

Comment:
When timeout happens, the thread is interrupted, but when the blocking call returns, the code continue to be executed. We want to prevent that.